src/ring_theory/graded_algebra/basic.lean

```diff
@@ -77,4 +80,101 @@ def graded_algebra.decompose : A ≃ₐ[R] ⨁ i, 𝒜 i := alg_equiv.symm
   (graded_algebra.decompose 𝒜).symm (direct_sum.of _ i x) = x :=
 direct_sum.submodule_coe_alg_hom_of 𝒜 _ _
 
+/-- The projection maps of graded algebra-/
+def graded_algebra.proj (𝒜 : ι → submodule R A) [graded_algebra 𝒜] (i : ι) : A →ₗ[R] A :=
+(𝒜 i).subtype.comp $
+  (dfinsupp.lapply i).comp $
+  (graded_algebra.decompose 𝒜).to_alg_hom.to_linear_map
+
+@[simp] lemma graded_algebra.proj_apply (i : ι) (r : A) :
+  graded_algebra.proj 𝒜 i r = (graded_algebra.decompose 𝒜 r : ⨁ i, 𝒜 i) i := rfl
+
+lemma graded_algebra.proj_mem (i : ι) (r : A) :
+  graded_algebra.proj 𝒜 i r ∈ 𝒜 i := (graded_algebra.decompose 𝒜 r i).2
+
```

Comment:
I'd be tempted to remove this, since `simp` know how to solve it now, and `proj A i r` is now reduced to `decompose A r i` anyway.